src/OCServo.h

```diff
@@ -0,0 +1,89 @@
+#ifndef OCSERVO_H
+#define OCSERVO_H
+
+#include <Arduino.h>
+#include <SoftwareSerial.h>
+
+#include "ROBSMotor.h"
+
+enum OCSMode {
+    OCS_SERVO,
+    OCS_MOTOR
+};
+
+enum OCSResponseLevel {
+    OCS_RESPOND_READ_ONLY,
+    OCS_RESPOND_ALL
+};
+
+typedef enum e_SerialPortType {
+    SERIAL_HARDWARE,
+    SERIAL_SOFTWARE
+} SerialPortType;
+
+struct OCSResponse {
+    byte prefix[2];
+    byte id;
+    byte length;
+    byte instruction;
+    int numberOfParameters;
+    byte parameters[16];
+    byte checksum;
+};
+
+class OCServo {
+    private:
+        byte id;
+        Stream *serial;
+        SerialPortType portType;
+
+        byte getChecksum(byte *data, int size);
+        byte baudRateToByte(long baudrate);
+        long byteToBaudRate(byte value);
+        OCSResponse bytesToResponse(byte *data, int size);
+        OCSResponse readResponse();
+
+    public:
+        OCServo(byte id, SoftwareSerial *serialPort);
+        OCServo(byte id, HardwareSerial *serialPort);
```

Comment:
The template-based approach eliminates the need for separate constructors for HardwareSerial and SoftwareSerial, streamlining the code by using a single constructor that deduces the port type automatically at compile time.

Also do not use pointer if not expect that value can be null pointer
```suggestion
    // Template-based constructor
    template<typename SerialType>
    OCServo(uint8_t id, SerialType &serialPort);
```

add bellow after class definition:

```
// Specialization for HardwareSerial
template<>
constexpr SerialPortType determinePortType<HardwareSerial>() {
    return SerialPortType::Hardware;
}

// Specialization for SoftwareSerial
template<>
constexpr SerialPortType determinePortType<SoftwareSerial>() {
    return SerialPortType::Software;
}

```